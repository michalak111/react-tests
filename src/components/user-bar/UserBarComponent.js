import React, {Component} from 'react'
import PropTypes from 'prop-types'
import './user-bar.scss'

class UserBarComponent extends Component {
  static propTypes = {
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired
  }
  render () {
    const {firstName, lastName, handleClick} = this.props
    return (
      <div className={'rp-user-bar card'} onClick={handleClick}>
        <p>First Name: <strong>{firstName}</strong></p>
        <p>Last Name: <strong>{lastName}</strong></p>
      </div>
    )
  }
}

export default UserBarComponent
