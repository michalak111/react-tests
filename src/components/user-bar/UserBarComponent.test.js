import React from 'react'
import {shallow} from 'enzyme'
import UserBarComponent from './UserBarComponent'
import renderer from 'react-test-renderer';

describe('<UserBarComponent />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<UserBarComponent handleClick={() => {}} firstName={'First Name'} lastName={'Last Name'}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should call props.handleClick when is clicked', () => {
    const fn = jest.fn()
    const userBar = shallow(<UserBarComponent handleClick={fn} firstName={'First Name'} lastName={'Last Name'}/>)
    userBar.simulate('click')
    expect(fn).toBeCalled()
  })
})
