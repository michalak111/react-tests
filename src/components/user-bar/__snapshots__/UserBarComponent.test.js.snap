// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserBarComponent /> renders correctly 1`] = `
<div
  className="rp-user-bar card"
  onClick={[Function]}
>
  <p>
    First Name: 
    <strong>
      First Name
    </strong>
  </p>
  <p>
    Last Name: 
    <strong>
      Last Name
    </strong>
  </p>
</div>
`;
